app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,126 @@
+samson.controller('ProjectRolesCtrl', function ($scope, $element, $filter, projectRolesService, messageCenterService) {
+    $scope.project_role = {};
+    $scope.roles = [];
+
+    $scope.initModel = function () {
+        $scope.project_role = {
+            id: getUniqueId(),
+            user_id: getUserId(),
+            project_id: getProjectId(),
+            role_id: getRoleId()
+        };
+
+        loadProjectRoles();
+    };
+
+    $scope.$watch('project_role.role_id', function (new_role_value, old_role_value) {
+        if (new_role_value !== old_role_value) {
+            if (exists($scope.project_role)) {
+                return updateProjectRole($scope.project_role, new_role_value);
+            }
+            else {
+                return createProjectRole($scope.project_role, new_role_value);
+            }
+        }
+    });
+
+    function loadProjectRoles() {
+        projectRolesService.loadProjectRoles().then(
+            function (response) {
+                $scope.roles = response.data;
+            }
+        );
+    }
+
+    function exists(project_role) {
+        return project_role.id;
+    }
+
+    function createProjectRole(project_role) {
+        projectRolesService.createProjectRole(project_role).then(
+            function (response) {
```

Comment:
@henders, got confused by your comment. Are you suggesting to use the $http success / error calbacks (which got deprecated) or something in the likes of:

```
projectRolesService.createProjectRole(project_role).then(
  function success(response) { ... },
  function error() { ... }
);
```

I just never saw it being used like this. I agree that it makes it more explicit, but... maybe just a comment in the first line of each function saying `//Success` or `//Error` ? :)

What you guys think?